### PR TITLE
FetchIt Challenge Package addition and minor melodic Moveit Fix

### PR DESCRIPTION
**Pull request by RDaneelOlivav**
Here we add the latest version of the files for the FetchIt! Challenge. All the models, environments and launches needed. We also added a minor fix for moveit of the melodic version that has different input parameters.

### CODE REVIEW

File: fetch_gazebo_demo/scripts/pick_place_demo.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def updateScene(self):
             self.scene.addSolidPrimitive(obj.object.name,
                                          obj.object.primitives[0],
                                          obj.object.primitive_poses[0],
-                                         wait = False)
+                                         use_service = False)
```
Comment (mikeferguson) [inline fetch_gazebo_demo/scripts/pick_place_demo.py:154]: same comment as on #26 - I think you actually want to just remove the "wait = False" -- the new service interface is quite speedy and very reliable (unlike the old "wait for sync").

Comment (RDaneelOlivav) [reply fetch_gazebo_demo/scripts/pick_place_demo.py:154]: I dont quite get you there. So you mean that its not necessary to indicate the 
**use_service = False**
? We just need to remove the wait?

And as for the addition of the fetchit_challenge package, everything ok there?

Comment (moriarty) [reply fetch_gazebo_demo/scripts/pick_place_demo.py:154]: You just need to remove the line, which will **use_service=True**
http://docs.ros.org/kinetic/api/moveit_python/html/classmoveit__python_1_1planning__scene__interface_1_1PlanningSceneInterface.html#a72a7f448769da906ae689420092ce4c4

The addition of the fetchit_challenge looks good enough for as the first pass, the diff is too large to really see in GitHub.

Comment (moriarty) [inline fetch_gazebo_demo/scripts/pick_place_demo.py:154]:
```suggestion
                                         )
```
@mikeferguson's suggestion to **use_service = True** makes sense to me.

GitHub suggestions don't support two line edits :disappointed:

File: fetch_gazebo_demo/scripts/pick_place_demo.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ def updateScene(self):
             self.scene.addSolidPrimitive(obj.name,
                                          obj.primitives[0],
                                          obj.primitive_poses[0],
-                                         wait = False)
+                                         use_service = False)
```
Comment (moriarty) [inline fetch_gazebo_demo/scripts/pick_place_demo.py]:
```suggestion
                                         )
```

**Review by moriarty — CHANGES_REQUESTED** [review]
Just the same thing pointed out by @mikeferguson.

GitHub doesn't support multi-line suggestions so if you want to use the web interface to accept the changes for this version- I'm sure the next time someone edits the files properly pep8 will likely complain about the `)` being all alone on a line.

**Comment by moriarty** [comment]
@RDaneelOlivav:

The tables will be (approx)

1. 80.0 cm high
2. 92.5 cm wide
3. 47.0 cm deep

I don’t know the exact surface colour yet. They’re currently particle board. So wood is close enough... but the final version will likely be white.

The tables will be filled in, with a hard material (so they can be used for navigation) and on that material will be the sponsor logos.

I will upload a photo of the first prototype

**Comment by moriarty** [comment]
img_7710
img_7711

This was one of our old demo warehouse shelves. I've cut it down to match the height of the Schunk machine.

They can be disassembled and then they flat-pack for shipping, so we will use the same shelves in Montreal.

We don't have hard covers for the sides yet, but lets assume they'll look similar to the Schunk machine. They will be made out of some stiff material (likely foam-core board) and attach sides with "velcro".
The other competition co-sponsors will place their logos on these shelf sides.

I just took a photo of the back to show that it was a shelf. But the version for gazebo does not need to model the inside, it can simply be a solid rectangle.

File: fetch_gazebo_demo/scripts/pick_place_demo.py (hunk 2 above)
Comment (RDaneelOlivav) [reply fetch_gazebo_demo/scripts/pick_place_demo.py]: Sorry for the late response, didnt get this update for some reason on my mail. So you want me to do two things basically change:
use_service = False --> use_service = True
And put the parenthesis in another line?

**Comment by RDaneelOlivav** [comment]
Oh sorry didnt get this update for some reason. Ok so let me get this straight:
The tables have to be designed in a way that they have the dimensions given and they look like the SHunk table there , right?

I'll start working on that and commit the changes as soon as possible

**Comment by moriarty** [comment]
Yes, don't worry about the actual sponsor logo matching, just make the tables rectangles for now.
You've already added the sponsor logs on the wall of the environment so that's fine.

The tables being solid, instead of having one thin leg in the middle, or four legs on in the corners, will help make it easier to use them for the laser scanners, and useful for navigation and alignment with the tables.

**Comment by RDaneelOlivav** [comment]
screenshot from 2019-02-20 21-47-19

screenshot from 2019-02-20 21-50-02

I've created new models for the tables based on the specifications given and tested also that the pick and place script demo ;).

The logos are just place holders, we afterward can place other logos there if it's needed.

Any other suggestion? As soon as we merge this branch I'll generate a new ROSject using the official fetch repositories with the latest changes, and from there we can keep on improving ;).

**Review by moriarty — APPROVED** [review]
LGTM. Thanks!

**Comment by moriarty** [comment]
table
something-is-not-correct

So, I did the review earlier it looks fine to merge and then we can fix the issues in future PRs.

But, the package.xml is missing some dependencies, and the CMakeLists.txt might not be installing meshes in the correct locations, when I ran it on my test machine, which has a bare ros installation, I the new tables had no surfaces, but they were there.

Also in future versions, and in the real environment we'll need to settle on colours.
The two screws will not be the same colour as the bin they're being picked from: that would be evil.

**Comment by RDaneelOlivav** [comment]
screenshot from 2019-02-21 10-39-50

Colour changed to a complementary colour, in theory it should make it as easier as it can get to differenciate between the bolts and the bin.

screenshot from 2019-02-21 10-39-44

I downloaded the code and compiled and it seemed to work for me. In theory the tag in package.xml in the fetchit_challenge package should do the trick to allow the system to find the models:
```
  <export>
    <gazebo_ros gazebo_model_path="${prefix}/models"/>
  </export>
```

So we can debug in another pull request what's happening there.

I also added dependencies to the fetchit_challenge package.

We should also check in another PR, the compilation of this whole repo because it needs to have precompiled the grasping_msgs package before compiling which is some of a inconveniece.

But yeah, lets merge and see what we organise tasks for debugging those issues

**Pull request merged**